src/query/api/v1/handler/prometheus/common.go

```diff
@@ -176,6 +176,35 @@ func parseTagCompletionQueries(r *http.Request) ([]string, error) {
 	return queries, nil
 }
 
+// ParseStartAndEnd parses start and end params from the request.
+func ParseStartAndEnd(
+	r *http.Request,
+	parseOpts xpromql.ParseOptions,
```

Comment:
why does this need the whole `parseOpts` instead of only the `NowFn()`?

Reply:
Figured this was cleaner than building a new `parseOpts` on each query; can go that route if preferred